api/envoy/config/common/matcher/v3/matcher.proto

```diff
@@ -16,6 +19,80 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Unified Matcher API]
 
+message MatchTree {
+  message MatchAction {
+    oneof action {
+      bool skip = 1;
+
+      string callback = 2;
+    }
+  }
+
+  message MatchLeaf {
+    message LeafMatcher {
+      // The predicate to evaluate.
+      oneof matcher {
+        MatchPredicate predicate = 1;
+
+        core.v3.TypedExtensionConfig typed_config = 2;
+      }
+    }
+
+    // A linear set of match conditions that will be evaluated in order.
+    // The action of the first matcher that evaluates to true
+    // decides the action that will be performed. If no match
+    // criteria is matched, the no_match_action action will be performed.
+    repeated LeafMatcher matchers = 1;
+
+    // The action to perform if the predicate evaluates to true.
+    MatchAction action = 2;
+
```

Comment:
Don't we want [(match, action)] at the leaf, rather than ([match],action)?

Reply:
Yeah I think that makes senses, I'll update